include/cppkafka/header_list.h

```diff
@@ -210,14 +209,16 @@ HeaderList<HeaderType>::HeaderList(size_t reserve)
 
 template <typename HeaderType>
 HeaderList<HeaderType>::HeaderList(rd_kafka_headers_t* handle)
-: handle_(handle, &rd_kafka_headers_destroy, &rd_kafka_headers_copy) { //if we own the header list, we clone it on copy
-
+: handle_(handle,
+          handle ? &rd_kafka_headers_destroy : nullptr,
+          handle ? &rd_kafka_headers_copy : nullptr) { //if we own the header list, we clone it on copy
```

Comment:
I'm not sure about this change. Why do we want to allow building a `HeaderList` on top of a null handle? This should probably assert that the handle isn't null. I presume you're using this in `Message::set_header_list`, right? I'd rather than one make sure it's not calling this with a null pointer rather than this one "adapting" to null pointers when in reality this shouldn't work.

Reply:
ok

Reply:
The thing is, this constructor will be called both by cppkafka and external users of the library. You want to be friendly to external users and not let their broken code compile (e.g. allowing doing `HeaderList(nullptr)` shouldn't work)

If there's 3 places inside cppkafka where there's the possibility of calling that with an null pointer, I'd rather add the checks in those 3 places rather than being slightly more flexible on the `HeaderList` API but allowing the user to create a header that contains nothing bypassing the default constructor.

Reply:
Yes, this is what I've done in the last commit.